test/model/typing_status_test.dart

```diff
@@ -0,0 +1,240 @@
+import 'dart:async';
+
+import 'package:checks/checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/typing_status.dart';
+
+import '../example_data.dart' as eg;
+import '../fake_async.dart';
+
+extension TypingStatusChecks on Subject<TypingStatus> {
+  Subject<Map<SendableNarrow, Map<int, Timer>>> get typistsByNarrow => has((x) => x.typistIdsByNarrow, 'typistsByNarrow');
+}
+
+extension TimerChecks on Subject<Timer> {
+  Subject<bool> get isActive => has((x) => x.isActive, 'isActive');
+}
+
+void main() {
+  late TypingStatus model;
+  late int notifiedCount;
+
+  void prepareModel() {
+    model = TypingStatus(
+      selfUserId: eg.selfUser.userId,
+      typingStartedExpiryPeriod: const Duration(milliseconds: 15000));
+    check(model).typistsByNarrow.isEmpty();
+    notifiedCount = 0;
+    model.addListener(() => notifiedCount += 1);
+  }
+
+  void checkNotNotified() {
+    check(notifiedCount).equals(0);
+  }
+
+  void checkNotifiedOnce() {
+    check(notifiedCount).equals(1);
+    notifiedCount = 0;
+  }
+
+  Subject<TypingStatus> checkTyping(SendableNarrow narrow, TypingOp op, User typist) {
+    model.handleTypingEvent(eg.typingEvent(narrow, op, typist.userId));
+    return check(model);
+  }
+
+  final stream = eg.stream();
+  final topicNarrow = TopicNarrow(stream.streamId, 'foo');
+
+  final dmNarrow = DmNarrow.withUser(eg.otherUser.userId, selfUserId: eg.selfUser.userId);
+  final groupNarrow = DmNarrow.withOtherUsers(
+    [eg.otherUser.userId, eg.thirdUser.userId], selfUserId: eg.selfUser.userId);
+
+  anyTimer(Subject<Object?> it) => it.isA<Timer>();
```

Comment:
Function definitions should have explicit return types, so that they don't look like function calls:
```suggestion
  void anyTimer(Subject<Object?> it) => it.isA<Timer>();
```